src/UKHO.S100PermitService.Common/Clients/ProductKeyServiceApiClient.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Text;
+
+namespace UKHO.S100PermitService.Common.Clients
+{
+    public class ProductKeyServiceApiClient : IProductKeyServiceApiClient
+    {
+        private readonly HttpClient _httpClient;
+
+        public ProductKeyServiceApiClient(IHttpClientFactory httpClientFactory)
+        {
+            _httpClient = httpClientFactory.CreateClient();
+        }
+
+        public async Task<HttpResponseMessage> CallProductKeyServiceApiAsync(string uri, HttpMethod httpMethod, string payLoad, string accessToken, string correlationId)
```

Comment:
The client should have one method per endpoint in the service implementation. The caller of this method should not have to construct the payload, specify the HTTP method - you shouldn't even care that it is calling a REST endpoint. Rename the method according to the endpoint on the PKS it is calling, and build the payload in this method from parameters passed in.

Reply:
Resolved